src/main/java/net/countercraft/movecraft/warfare/commands/AssaultCommand.java

```diff
@@ -114,8 +115,12 @@ public boolean onCommand(CommandSender commandSender, Command command, String s,
         final Long taskMaxDamages = (long) AssaultUtils.getMaxDamages(region);
         final Vector taskMin = min;
         final Vector taskMax = max;
-
-        // TODO: Make async
+        Assault assault = new Assault(region, taskPlayer, taskWorld, System.currentTimeMillis(), taskMaxDamages, taskMin, taskMax);
+        final AssaultStartEvent event = new AssaultStartEvent(assault);
```

Comment:
The actual assault isn't created or started until after the below task is run.  This is something I very much would like to change, but I foresee it as being a large change to the codebase.